pkg/fatfs/fatfs_vfs/fatfs_vfs.c

```diff
@@ -90,6 +90,9 @@ static int _format(vfs_mount_t *mountp)
 
     /* make sure the volume has been initialized */
     if (_init(mountp)) {
+        if (!IS_USED(CONFIG_FATFS_FORMAT_ALLOC_STATIC)) {
+            free(work);
+        }
```

Comment:
Ah sorry, we would have to always use the C-conditionals in that function in order for the compiler to catch that `work` might be dynamically allocated - since we are not, better follow in the same style.

```suggestion
#if !CONFIG_FATFS_FORMAT_ALLOC_STATIC
        free(work);
#endif
```